compiler/parse.py

```
@@ -587,15 +587,24 @@ def __init__(self, debug=False, logger=None, optimize=True,
         else:
             self.logger = logger
 
-        # Explicitly silence warnings about unused rules when
-        # starting from a state other than the default.
-        errorlog = None if start == 'program' else yacc.NullLogger()
+        if start == 'program':
+            errorlog = None
+            yaccfile = 'parsetab'
+        else:
+            # Explicitly silence warnings about unused rules when
+            # starting from a state other than the default. In addition,
+            # send parser cache to a special file, to avoid conflicts and
+            # make cleaning easy
+            errorlog = yacc.NullLogger()
+            yaccfile = ("%s_%s") % ('aux', start)
```

This logger is used internally during the construction of the parsing tables. If `start` is different than `program`, many grammar rules will be unreachable. This causes `PLY` to produce tons of warnings. But unreachable grammar rules should only happen during testing. We are explicitly silencing the warnings using a `logger` that silently drops each error it receives.

This practice is outlined in the PLY's documentation. Search for `NullLogger`.